docs/source/extending/mynorm.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+from numba import njit, types
+from numba.extending import overload, register_jitable
+from numba.errors import TypingError
+
+import scipy.linalg
+
+
+@register_jitable
+def _oneD_norm_2(a):
+    # re-usable implementation of the 2-norm
+    val = np.abs(a)
+    return np.sqrt(np.sum(val * val))
+
+
+@overload(scipy.linalg.norm)
+def jit_norm(a, ord=None):
+    if isinstance(ord, types.Optional):
+        ord = ord.type
+    # Reject non integer, floating-point or None types for ord
+    if not isinstance(ord, (types.Integer, types.Float, types.NoneType)):
+        raise TypingError("'ord' must be either integer or floating-point")
+    # Reject non-ndarray types
+    if not isinstance(a, types.Array):
+        raise TypingError("Only accepts NumPy ndarray")
+    # Reject ndarrays with non integer or floating-point dtype
+    if not isinstance(a.dtype, (types.Integer, types.Float)):
+        raise TypingError("Only integer and floating point types accpeted")
+    # Reject ndarrays with unsupported dimensionality
+    if a.ndim not in [1, 2]:
+        raise TypingError('3D and beyond are not allowed')
+    # Implementation for vectors
```

Comment:
This comment applies to the next branch.

Reply:
Good catch!